flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/TestingPhysicalSlotProvider.java

```diff
@@ -69,71 +63,43 @@ private TestingPhysicalSlotProvider(SlotSharingExecutionSlotAllocatorTest.Physic
 		this.requests = new HashMap<>();
 		this.responses = new HashMap<>();
 		this.cancellations = new HashMap<>();
-
-		this.availableSlotCount = availableSlotCount;
-		this.allocateLatch = new CountDownLatch(availableSlotCount);
 	}
 
 	@Override
 	public CompletableFuture<PhysicalSlotRequest.Result> allocatePhysicalSlot(PhysicalSlotRequest physicalSlotRequest) {
 		SlotRequestId slotRequestId = physicalSlotRequest.getSlotRequestId();
 		requests.put(slotRequestId, physicalSlotRequest);
-		CompletableFuture<TestingPhysicalSlot> resultFuture = new CompletableFuture<>();
-		responses.put(slotRequestId, resultFuture);
-
-		switch (physicalSlotFutureCompletion) {
-			case SUCCESS:
-				if (availableSlotCount > 0) {
-					completePhysicalSlotFutureFor(slotRequestId, new AllocationID());
-				} else {
-					resultFuture.completeExceptionally(new NoResourceAvailableException());
-				}
-				break;
-			case FAILURE:
-				resultFuture.completeExceptionally(new FlinkException("Test failure."));
-				break;
+		CompletableFuture<TestingPhysicalSlot> resultFuture;
+		try {
+			resultFuture = physicalSlotCreator.apply(
+				taskManagerLocation,
+				taskManagerGateway,
+				physicalSlotRequest.getSlotProfile().getPhysicalSlotResourceProfile());
+		} catch (Throwable t) {
+			resultFuture = FutureUtils.completedExceptionally(t);
```

Comment:
Given that `physicalSlotCreator` can return a `CompletableFuture`, why do we need to catch exceptions here? Couldn't the creator return an exceptionally completed future? If the creator only returns a `TestingPhysicalSlot`, then this would be needed.

Reply:
Yeah, you're right. I should have used the `TriFunction` right away.